src/oop/object_independence/hard/q2/MainTest.java

```diff
@@ -0,0 +1,299 @@
+package oop.object_independence.hard.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String objectClassName = "Fish";
+    private final String testClassName = "TestAquarium";
+    private final String stringAttributeName1 = "colour";
+    private final String stringAttributeName2 = "gender";
+    private final String intAttributeName1 = "age";
+    private final String getStringAttributeName1 = "getColour";
+    private final String getStringAttributeName2 = "getGender";
+    private final String getIntAttributeName1 = "getAge";
+    private final String setStringAttributeName1 = "setColour";
+    private final String setStringAttributeName2 = "setGender";
+    private final String setIntAttributeName1 = "setAge";
+    private final String testStringAttributeName1a = "Blue";
+    private final String testStringAttributeName2a = "White";
+    private final String testStringAttributeName3a = "Red";
+    private final String testStringAttributeName1b = "Male";
+    private final String testStringAttributeName2b = "Female";
+    private final String testStringAttributeName3b = "Female";
+    private final int testIntAttributeName1 = 2;
+    private final int testIntAttributeName2 = 4;
+    private final int testIntAttributeName3 = 3;
+    private final int testIntAttributeNameUpdate = 10;
+    private ObjectTest classInstance;
+    private ObjectTest testInstance;
+
+    @BeforeEach
+    public void setup() {
+        String objectClassString = "oop.object_independence.hard.q2." + objectClassName;
+        String testClassString = "oop.object_independence.hard.q2." + testClassName;
+        classInstance = new ObjectTest(objectClassString);
+        testInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void fishClassHasCorrectAttributes() {
+        String missingAttributeMessage = "Your %s class is missing the %s attribute. Please make sure you have added it and that it is spelled correctly.";
+        assertTrue(classInstance.hasField(stringAttributeName1, String.class),
```

Comment:
This `hasField` call is checking not just the name, but also the class already. Please check them separately.